utils/merkletrie/fsnoder/new.go

```diff
@@ -0,0 +1,193 @@
+package fsnoder
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie/hasher"
```

Comment:
Separate package for hasher? I think the same as for `noder`, looks like overdoing it.